bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/handler/VolvoOnCallBridgeHandler.java

```diff
@@ -58,144 +42,60 @@
+    private final HttpClient httpClient;
 
-    private @NonNullByDefault({}) CustomerAccounts customerAccount;
+    private @Nullable VocHttpApi api;
 
-    public VolvoOnCallBridgeHandler(Bridge bridge) {
+    public VolvoOnCallBridgeHandler(Bridge bridge, Gson gson, HttpClient httpClient) {
         super(bridge);
-
-        httpHeader.put("cache-control", "no-cache");
-        httpHeader.put("content-type", JSON_CONTENT_TYPE);
-        httpHeader.put("x-device-id", "Device");
-        httpHeader.put("x-originator-type", "App");
-        httpHeader.put("x-os-type", "Android");
-        httpHeader.put("x-os-version", "22");
-        httpHeader.put("Accept", "*/*");
-
-        gson = new GsonBuilder()
-                .registerTypeAdapter(ZonedDateTime.class,
-                        (JsonDeserializer<ZonedDateTime>) (json, type, jsonDeserializationContext) -> ZonedDateTime
-                                .parse(json.getAsJsonPrimitive().getAsString().replaceAll("\\+0000", "Z")))
-                .registerTypeAdapter(OpenClosedType.class,
-                        (JsonDeserializer<OpenClosedType>) (json, type,
-                                jsonDeserializationContext) -> json.getAsBoolean() ? OpenClosedType.OPEN
-                                        : OpenClosedType.CLOSED)
-                .registerTypeAdapter(OnOffType.class,
-                        (JsonDeserializer<OnOffType>) (json, type,
-                                jsonDeserializationContext) -> json.getAsBoolean() ? OnOffType.ON : OnOffType.OFF)
-                .create();
+        this.gson = gson;
+        this.httpClient = httpClient;
     }
 
     @Override
     public void initialize() {
         logger.debug("Initializing VolvoOnCall API bridge handler.");
-        VolvoOnCallBridgeConfiguration configuration = getConfigAs(VolvoOnCallBridgeConfiguration.class);
+        ApiBridgeConfiguration configuration = getConfigAs(ApiBridgeConfiguration.class);
 
-        httpHeader.setProperty("Authorization", configuration.getAuthorization());
         try {
-            customerAccount = getURL(SERVICE_URL + "customeraccounts/", CustomerAccounts.class);
-            if (customerAccount.username != null) {
-                updateStatus(ThingStatus.ONLINE);
+            api = new VocHttpApi(configuration, gson, httpClient);
+            CustomerAccounts account = api.getURL("customeraccounts/", CustomerAccounts.class);
+            if (account.username != null) {
+                updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, account.username);
             } else {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                        "Incorrect username or password");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Incorrect login credentials");
             }
-        } catch (JsonSyntaxException | VolvoOnCallException e) {
+        } catch (Exception e) {
```

Comment:
Please catch specific exceptions, or at the very least catch RuntimeException instead.